mongosql/src/schema/definitions.rs

```diff
@@ -1493,7 +1493,7 @@ impl Schema {
     pub fn union(&self, other: &Schema) -> Schema {
         use std::cmp::Ordering;
         use Schema::*;
-        let (left, right) = (Self::simplify(self), Self::simplify(other));
+        let (left, right) = (self.clone(), other.clone());
```

Comment:
These aren't necessarily "new" clones, since ultimately `Schema::simplify` clones the entire `Schema` tree.

We may be able to rework this library to be better suited to avoid cloning, but as it stands right now we just swap between `&Schema` and `Schema` too frequently and liberally.

Maybe the memory team is considering tackling this library, or maybe they already ruled out the need to do it. Either way, these clones are effectively equivalent to the cloning `simplify` was already doing.